RTXOff/rtxoff_mempool.cpp

```diff
@@ -0,0 +1,666 @@
+
+    return mp->mp_info.max_blocks;
+}
+
+/// Get memory block size in a Memory Pool.
+/// \note API identical to osMemoryPoolGetBlockSize
+static uint32_t svcRtxMemoryPoolGetBlockSize(osMemoryPoolId_t mp_id) {
+    auto *mp = reinterpret_cast<osRtxMemoryPool_t *>(mp_id);
+    // Check parameters
+    if ((mp == nullptr) || (mp->id != osRtxIdMemoryPool)) {
+        // add event
+        // EvrRtxMemoryPoolGetBlockSize(mp, 0U);
+        //lint -e{904} "Return statement before end of function" [MISRA Note 1]
+        return 0U;
+    }
+
+    // add event
+    // EvrRtxMemoryPoolGetBlockSize(mp, mp->mp_info.block_size);
+
+    return mp->mp_info.block_size;
+}
+
+/// Get number of memory blocks used in a Memory Pool.
+/// \note API identical to osMemoryPoolGetCount
+static uint32_t svcRtxMemoryPoolGetCount(osMemoryPoolId_t mp_id) {
+    auto *mp = reinterpret_cast<osRtxMemoryPool_t *>(mp_id);
+    // Check parameters
+    if ((mp == nullptr) || (mp->id != osRtxIdMemoryPool)) {
+        // add event
+        // EvrRtxMemoryPoolGetCount(mp, 0U);
+        //lint -e{904} "Return statement before end of function" [MISRA Note 1]
+        return 0U;
+    }
+
+    // add event
+    // EvrRtxMemoryPoolGetCount(mp, mp->mp_info.used_blocks);
+
+    return mp->mp_info.used_blocks;
+}
+
+/// Get number of memory blocks available in a Memory Pool.
+/// \note API identical to osMemoryPoolGetSpace
+static uint32_t svcRtxMemoryPoolGetSpace(osMemoryPoolId_t mp_id) {
+    auto *mp = reinterpret_cast<osRtxMemoryPool_t *>(mp_id);
+    // Check parameters
+    if ((mp == nullptr) || (mp->id != osRtxIdMemoryPool)) {
+        // add event
+        // EvrRtxMemoryPoolGetSpace(mp, 0U);
+        //lint -e{904} "Return statement before end of function" [MISRA Note 1]
+        return 0U;
+    }
+
+    // add event
+    // EvrRtxMemoryPoolGetSpace(mp, mp->mp_info.max_blocks - mp->mp_info.used_blocks);
+
+    return (mp->mp_info.max_blocks - mp->mp_info.used_blocks);
+}
+
+/// Delete a Memory Pool object.
+/// \note API identical to osMemoryPoolDelete
+static osStatus_t svcRtxMemoryPoolDelete(osMemoryPoolId_t mp_id) {
+    auto *mp = reinterpret_cast<osRtxMemoryPool_t *>(mp_id);
+    osRtxThread_t *thread;
+
+    // Check parameters
+    if ((mp == nullptr) || (mp->id != osRtxIdMemoryPool)) {
+        // add event
+        // EvrRtxMemoryPoolError(mp, (int32_t) osErrorParameter);
+        //lint -e{904} "Return statement before end of function" [MISRA Note 1]
+        return osErrorParameter;
+    }
+
+    // Unblock waiting threads
+    if (mp->thread_list != nullptr) {
+        do {
+            thread = osRtxThreadListGet(reinterpret_cast<osRtxObject_t *>(mp));
+            osRtxThreadWaitExit(thread, 0U, false);
+        } while (mp->thread_list != nullptr);
+
+        // at this point a new thread might potentially take over
+        ThreadDispatcher::instance().dispatch(nullptr);
+
+        // TODO unclear whether the above is enough
```

Comment:
@multiplemonomials not sure if this is necessary - I simply copied it from rtxoff_semaphore.cpp

Reply:
Looks right too me.  Good job figuring this out, it's one of the less obvious parts.